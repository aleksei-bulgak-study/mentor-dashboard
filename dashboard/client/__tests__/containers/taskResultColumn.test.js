import React from 'react';
import { shallow } from 'enzyme';
import TaskResultColumn from '../../src/containers/taskResultColumn';

describe('TaskResultColumn', () => {
  it('test snapshot does not changed', () => {
    // given
    const props = {
      status: 'Checked',
      result: { mark: 145 },
    };
    // when
    const wrapper = shallow(<TaskResultColumn {...props} />);
    // then
    expect(wrapper).toMatchSnapshot();
  });

  it('test snapshot does not changed when status is invalid', () => {
    // given
    const props = {
      status: 'Invalid',
      result: { pr: 'https://example.com', mark: 145 },
    };
    // when
    const wrapper = shallow(<TaskResultColumn {...props} />);
    // then
    expect(wrapper).toMatchSnapshot();
  });

  it('test snapshot does not changed when result is empty', () => {
    // given
    const props = {
      status: 'Checking',
    };
    // when
    const wrapper = shallow(<TaskResultColumn {...props} />);
    // then
    expect(wrapper).toMatchSnapshot();
  });


  it('test snapshot does not changed when work is in progress', () => {
    // given
    const props = {
      status: 'In Progress',
    };
    // when
    const wrapper = shallow(<TaskResultColumn {...props} />);
    // then
    expect(wrapper).toMatchSnapshot();
  });
});
