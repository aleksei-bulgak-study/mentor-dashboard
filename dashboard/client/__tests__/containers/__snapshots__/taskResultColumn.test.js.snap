// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskResultColumn test snapshot does not changed 1`] = `
<TableItem
  color="green"
  link=""
  mark={145}
/>
`;

exports[`TaskResultColumn test snapshot does not changed when result is empty 1`] = `
<TableItem
  color="pink"
  link="#"
  mark={0}
/>
`;

exports[`TaskResultColumn test snapshot does not changed when status is invalid 1`] = `
<TableItem
  color="green"
  link="https://example.com"
  mark={145}
/>
`;

exports[`TaskResultColumn test snapshot does not changed when work is in progress 1`] = `
<TableItem
  color="yellow"
  link="#"
  mark={0}
/>
`;
